charts/coms/templates/secret.yaml

```diff
@@ -13,26 +21,62 @@
 apiVersion: v1
 kind: Secret
 metadata:
+  {{- if not .Values.config.releaseScoped }}
   annotations:
     "helm.sh/resource-policy": keep
+  {{- end }}
   name: {{ $bSecretName }}
   labels: {{ include "coms.labels" . | nindent 4 }}
 type: kubernetes.io/basic-auth
 data:
-  password: {{ $bPassword }}
-  username: {{ $bUsername }}
+  password: {{ .Values.basicAuthSecretOverride.password | default $bPassword | quote }}
+  username: {{ .Values.basicAuthSecretOverride.username | default $bUsername | quote }}
 {{- end }}
 {{- if not $pSecret }}
 ---
 apiVersion: v1
 kind: Secret
 metadata:
+  {{- if not .Values.config.releaseScoped }}
   annotations:
     "helm.sh/resource-policy": keep
+  {{- end }}
   name: {{ $pSecretName }}
   labels: {{ include "coms.labels" . | nindent 4 }}
 type: kubernetes.io/basic-auth
 data:
-  password: {{ $pPassword }}
-  username: {{ $pUsername }}
+  password: {{ .Values.dbSecretOverride.password | default $pPassword | quote }}
+  username: {{ .Values.dbSecretOverride.username | default $pUsername | quote }}
+{{- end }}
+{{- if and (not $oSecret) (and .Values.objectStorageSecretOverride.password .Values.objectStorageSecretOverride.username) }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  {{- if not .Values.config.releaseScoped }}
+  annotations:
+    "helm.sh/resource-policy": keep
+  {{- end }}
+  name: {{ $oSecretName }}
+  labels: {{ include "coms.labels" . | nindent 4 }}
+type: kubernetes.io/basic-auth
+data:
+  password: {{ .Values.objectStorageSecretOverride.password | default $oPassword | quote }}
+  username: {{ .Values.objectStorageSecretOverride.username | default $oUsername | quote }}
```

Comment:
We can drop the default generated fallback clauses ans just pipe quote the values here.